sklearn/_fast_gradient_boosting/tests/test_loss.py

```diff
@@ -0,0 +1,192 @@
+import numpy as np
+from numpy.testing import assert_almost_equal
+import scipy
+from scipy.optimize import newton
+from sklearn.utils import assert_all_finite
+import pytest
+
+from sklearn._fast_gradient_boosting.loss import _LOSSES
+from sklearn._fast_gradient_boosting.types import Y_DTYPE
+
+
+def get_derivatives_helper(loss):
+    """Return get_gradients() and get_hessians() functions for a given loss.
+    """
+
+    def get_gradients(y_true, raw_predictions):
+        # create gradients and hessians array, update inplace, and return
+        shape = raw_predictions.shape[0] * raw_predictions.shape[1]
+        gradients = np.empty(shape=shape, dtype=Y_DTYPE)
+        hessians = np.empty(shape=shape, dtype=Y_DTYPE)
+        loss.update_gradients_and_hessians(gradients, hessians, y_true,
+                                           raw_predictions)
+
+        if loss.__class__ is _LOSSES['least_squares']:
+            gradients *= 2  # ommitted a factor of 2 to be consistent with LGBM
+
+        return gradients
+
+    def get_hessians(y_true, raw_predictions):
+        # create gradients and hessians array, update inplace, and return
+        shape = raw_predictions.shape[0] * raw_predictions.shape[1]
+        gradients = np.empty(shape=shape, dtype=Y_DTYPE)
+        hessians = np.empty(shape=shape, dtype=Y_DTYPE)
+        loss.update_gradients_and_hessians(gradients, hessians, y_true,
+                                           raw_predictions)
+
+        if loss.__class__ is _LOSSES['least_squares']:
+            # hessians aren't updated because they're constant
+            hessians = np.full_like(y_true, fill_value=2)
+
+        return hessians
+
+    return get_gradients, get_hessians
+
+
+@pytest.mark.parametrize('loss, x0, y_true', [
+    ('least_squares', -2., 42),
+    ('least_squares', 117., 1.05),
+    ('least_squares', 0., 0.),
+    # I don't understand why but y_true == 0 fails :/
+    # ('binary_crossentropy', 0.3, 0),
```

Comment:
@ogrisel any idea why this could happen?

Reply:
I agree... I'm not super enthusiastic about this test TBH. To me `test_numerical_gradients` is more than enough.

@ogrisel do you really want to keep this test and if yes, do you have any clue why this fails now with y_true=0? This exact same test passes on pygbm, but as @thomasjpfan it's not suprising that newton doesn't converge given the shape of the function.